pep-0722.rst

```diff
@@ -0,0 +1,565 @@
+a metadata block, and tools MUST ignore it. The block type is all of the text on
+the initial line, up to the colon. There must be no whitespace before the colon.
+The initial line MAY contain text after the colon. How this is interpreted
+depends on the block type. Block types MUST be treated as case sensitive.
+
+The interpretation of any lines in a metadata block after the initial
+identifying line, is defined by the type of block.
+
+Tools MUST ignore any blocks with types they do not handle.
+
+Block types starting with the characters ``X-`` are reserved for the user, and
+MUST NOT be given a meaning in any future standard.
+
+Otherwise, the only defined block type is ``Script Dependencies``. For this
+block type,
+
+1. Text after the colon on the initial line is NOT allowed.
+2. All subsequent lines MUST contain :pep:`508` requirement
+   specifiers, one per line.
+
+There MUST only be a single ``Script Dependencies`` block in the file. Tools
+consuming dependency data MAY simply process the first such block found. This
+avoids the need for tools to process more data than is necessary. Stricter tools
+MAY, however, fail with an error if multiple script dependency blocks are
+present.
+
+Consumers MUST validate that at a minimum, all dependencies start with a
+``name`` as defined in :pep:`508`, and they MAY validate that all dependencies
+conform fully to :pep:`508`. They MUST fail with an error if they find an
+invalid specifier.
+
+Example
+-------
+
+The following is an example of a script with an embedded dependency block::
+
+    # In order to run, this script needs the following 3rd party libraries
+    #
+    ## Script Dependencies:
+    ##    requests
+    ##    rich
+
+    import requests
+    from rich.pretty import pprint
+
+    resp = requests.get("https://peps.python.org/api/peps.json")
+    data = resp.json()
+    pprint([(k, v["title"]) for k, v in data.items()][:10])
+
+
+Backwards Compatibility
+=======================
+
+As metadata blocks take the form of a structured comment, they can be added
+without altering the meaning of existing code.
+
+It is possible that a comment may already exist which matches the form of a
+metadata block. While the use of a double ``#`` prefix is intended to minimise
+this risk, it is still possible.
+
+Because tools must ignore unrecognised metadata types, the only potential issue
+we need to consider is script dependencies. In that case, a tool might read the
+wrong dependencies. In practice, though, this is unlikely to happen, as (a) the
+header text (``Script Dependencies:``) is fairly unusual, and (b) any following
+lines are unlikely to conform to :pep:`508` unless they *are* dependencies.
+
+In the rare case where an existing comment would be interpreted incorrectly as a
+dependency block, this can be addressed by adding an actual dependency block
+(which can be empty if the script has no dependencies) earlier in the code.
```

Comment:
This would run afoul of the earlier rule, “There MUST only be a single ``Script Dependencies`` block in the file.”

Reply:
Good point. I'll relax the MUST, as all it's actually intended to say is that tools are allowed to ignore anything except the first one.